docs/source/contributor-guide/roadmap.md

```diff
@@ -62,14 +63,35 @@ For more information:
 ### Discussing New Features
 
 If you plan to work on a new feature that doesn't have an existing ticket, it is
-a good idea to open a ticket to discuss the feature. Advanced discussion often
-helps avoid wasted effort by determining early if the feature is a good fit for
-DataFusion before too much time is invested. Discussion on a ticket can help
-gather feedback from the community and is likely easier to discuss than a 1000
-line PR.
-
-If you open a ticket and it doesn't get any response, you can try `@`-mentioning
-recently active community members in the ticket to get their attention.
+a good idea to open one for discussion. Advanced discussion helps avoid wasted
+effort by determining if the feature is a good fit for DataFusion before too
+much time is invested. Discussion on a ticket can help gather feedback from the
+community and is likely easier to discuss than a 1000 line PR.
+
+Maintainers will mark major proposals as `PROPOSED EPIC` to make them more
+visible, but we are very limited on review bandwidth. If you open a ticket and it
+doesn't get any response, try `@`-mentioning recently active community members
+in the ticket, or [posting to the mailing list or Discord](communication.md).
+
+### Sponsoring Maintainers
+
+We have found that most successful epics have one or more "sponsoring
+maintainers", a committer ([see here for current list]) who take the lead on reviewing and committing PRs, help with design,
+and coordinate and communicate with the community. Without a sponsoring
+maintainer, good PRs have remained unreviewed for a very long time.
+
+While sponsoring maintainer is not an official designated role, it is a helpful
+concept to understand how features (especially major ones) are implemented in
+DataFusion. Helping complete an epic is a significant time commitment, so
+maintainers are more likely to sponsor features they are particularly interested
+in or align with their use of DataFusion.
+
+If you are willing to be a sponsoring maintainer for a feature, please say so
```

Comment:
Thanks @alamb please help me to understand `Sponsoring Maintainters`. Are you suggesting to create a pool of active maintainers for specific feature, like XYZ, ABC is supporting Join development, QWE for planning, ZXC for optimizer work? 
So the PR author can tag all of them and depending on availability the maintainers can help review.

Is my understanding correct?

Reply:
I was trying to make it clear that any project that doesn't have an active maintainer that is willing to help move it along to completion is unlikely to succeed. Maybe this is obvious but I think it might help to make it explicit. 

I don't think we have a formal process (yet) to find such a maintainer other than the normal course of comments, etc. Clearly I play this role for many of the projects, but I am hoping to move to a place where other committers can also drive move projects too.

Hopefully that makes sense

Reply:
I'll try and reword this a bit for clarity too